chainermn/datasets/scatter_dataset.py

```diff
@@ -1,42 +1,85 @@
-import re
+import pickle
 import warnings
 
 import chainer.datasets
 import numpy
 
 
-class DataSizeError(RuntimeError):
-    def __init__(self, ds_size, pickled_size):
-        msg = """The dataset was too large to be scattered using MPI.
+class DataSizeError(object):
```

Comment:
In my environment of Python 3.6.0, the following error is raised in tset_dataset.py.

```
======================================================================
ERROR: test_scatter_large_dataset (test_dataset.TestDataset)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/kfukuda/chainermn/tests/test_dataset.py", line 102, in test_scatter_large_dataset
    lambda: self.scatter_large_data(comm_type))
  File "/home/kfukuda/.pyenv/versions/anaconda3-4.3.1/lib/python3.6/unittest/case.py", line 728, in assertRaises
    return context.handle('assertRaises', args, kwargs)
  File "/home/kfukuda/.pyenv/versions/anaconda3-4.3.1/lib/python3.6/unittest/case.py", line 158, in handle
    (name, self._base_type_str))
TypeError: assertRaises() arg 1 must be an exception type or tuple of exception types
```
I guess this is because DataSizeError is not inherited from RuntimeError or other exception classes.
I'm not sure if it is a version-dependent error, but could you look at it ?

Reply:
This is not because DataSizeError but the test does not fail any more! I pushed a test fix. Also I Changed DataSizeError to inherit RuntimeError which is trivial. It's left for applications that has `except DataSizeError:` clause which is not called any more.